src/settings.cpp

```diff
@@ -557,6 +560,19 @@ void read_settings_xml(pugi::xml_node root)
     }
     model::external_sources.push_back(make_unique<FileSource>(path));
   }
+  std::vector <double> source_strength ;
+  for (auto& s : model::external_sources)
+  {
+    source_strength.push_back(s->strength()); //is that strength normalized? if yes then it can be used as a pdf 
```

Comment:
Good question about normalization.  I expect this to return the value that the user provided to define the relative normalization of all the sources they have specified.  The current algorithm (source.cpp L#613-622) effectively normalizes this list during sampling.  I think that you can use the non-normalized list to correctly/safely initialize the `DiscreteIndex`, but it may be worth confirming that by looking at it's code/comments.